lib/modules/manager/gleam/artifacts.ts

```diff
@@ -0,0 +1,92 @@
+import is from '@sindresorhus/is';
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import {
+  deleteLocalFile,
+  findLocalSiblingOrParent,
+  readLocalFile,
+  writeLocalFile,
+} from '../../../util/fs';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+export async function updateArtifacts(
+  updateArtifact: UpdateArtifact,
+): Promise<UpdateArtifactsResult[] | null> {
+  const { packageFileName, updatedDeps, newPackageFileContent, config } =
+    updateArtifact;
+  logger.debug(`gleam.updateArtifacts(${packageFileName})`);
+  const isLockFileMaintenance = config.updateType === 'lockFileMaintenance';
+
+  if (is.emptyArray(updatedDeps) && !isLockFileMaintenance) {
+    logger.debug('No updated gleam deps - returning null');
+    return null;
+  }
+
+  // Find the first gleam dependency in order to handle mixed manager updates
+  const lockFileName = await findLocalSiblingOrParent(
+    packageFileName,
+    'manifest.toml',
+  );
+
+  if (!lockFileName) {
+    logger.debug(`No ${lockFileName} found`);
+    return null;
+  }
+
+  const oldLockFileContent = await readLocalFile(lockFileName);
+  if (!oldLockFileContent) {
+    logger.debug(`No ${lockFileName} found`);
+    return null;
+  }
+
+  try {
+    await writeLocalFile(packageFileName, newPackageFileContent);
+    if (isLockFileMaintenance) {
+      await deleteLocalFile(lockFileName);
+    }
+
+    const execOptions: ExecOptions = {
+      cwdFile: packageFileName,
+      docker: {},
+      toolConstraints: [
+        {
+          toolName: 'gleam',
```

Comment:
Needs adding containerbase/base repo

Reply:
huh. in local it ran great and fast... guessing it didnt actually use containers 😄 .

will drop a cross ref when ready

Reply:
https://github.com/containerbase/base/pull/2018